client/my-sites/plans/jetpack-plans/selector.tsx

```diff
@@ -188,6 +198,8 @@ const SelectorPage: React.FC< SelectorPageProps > = ( {
 					options={ { useJetpackGoogleAnalytics: ! isJetpackCloud() } }
 				/>
 
+				<LicensingActivationBanner siteId={ siteId } />
```

Comment:
You're missing checking the `siteId && enableUserLicensesDialog`. Without that, the banner is always rendered.